crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1435,10 +1485,10 @@ impl<'db> TypeInferenceBuilder<'db> {
                 Type::Tuple(tuple) => UnionType::from_elements(
                     self.db,
                     tuple.elements(self.db).iter().map(|ty| {
-                        ty.into_class_literal()
-                            .map_or(todo_type!(), |ClassLiteralType { class }| {
-                                Type::instance(class)
-                            })
+                        ty.into_class_literal().map_or(
+                            todo_type!("exception type"),
+                            |ClassLiteralType { class }| Type::instance(class),
+                        )
```

Comment:
I think we could actually resolve this TODO now that we support `type[T]`. But not something we need to address in this PR, obviously